fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/StringArithmetic.java

```diff
@@ -406,12 +406,8 @@ public static Expression characterLength(StringLikeLiteral first) {
         return new IntegerLiteral(first.getValue().length());
     }
 
-    private static boolean isSeparator(char c) {
-        if (".$|()[{^?*+\\".indexOf(c) == -1) {
-            return false;
-        } else {
-            return true;
-        }
+    private static boolean isAlphabetic(char c) {
+        return String.valueOf(c).matches("[A-Za-z]");
```

Comment:
should consider such as `ÀÇ`, u should use `\P{L}`.
refer: https://www.regular-expressions.info/unicode.html#prop
notice, BE has bug when process UTF-8 in some functions, so should not follow BE's behavior